src-v2/Judge/NTS.Judge/Adapters/Behinds/StartlistBehind.cs

```diff
@@ -1,63 +1,45 @@
-﻿using System.Diagnostics;
-using System.Timers;
-using Not.Application.Adapters.Behinds;
+﻿using Not.Application.Adapters.Behinds;
 using Not.Application.Ports.CRUD;
 using Not.Exceptions;
-using NTS.Domain;
+using Not.Startup;
 using NTS.Domain.Core.Entities;
 using NTS.Domain.Core.Objects;
-using NTS.Domain.Objects;
+using NTS.Domain.Core.Objects.Payloads;
 using NTS.Judge.Blazor.Ports;
-using NTS.Judge.Factories;
 
 namespace NTS.Judge.Adapters.Behinds;
 
-public class StartlistBehind : ObservableBehind, IStartlistBehind
+public class StartlistBehind : ObservableBehind, IStartupInitializer, IStartlistBehind
```

Comment:
I see you've implemented in in a similar way to Handouts, but I think we can do better in retrospective, especially since we have `StartList` object already exisitng in our Domain Model:
1. Subscribe/Unsubscribe to `PhaseCompleted` in the StartList itself
2. Expose a `Changed` Event in Startlist which emits when a Start has been added
3. Subscribe to that event in `StartlistBehind` and invoke `EmitChanged`. You can even include the Action to subscribe to as a `Startlist` ctor overload to indicate that events based communication is used.

This approach will encapsulate the entirety of the business flow within the Domain which is what we're aiming to do. 

If Startlist is not created and as such not subscribed to `PhaseCompleted` it's still a non-issue, because it will be generated from existing Participations when necessary.

Reply:
If this approach works well, as I think it will, we can convert the Handouts as well.